service/controller/resource/cloudconfigblob/create.go

```diff
@@ -0,0 +1,143 @@
+	"github.com/giantswarm/microerror"
+	"github.com/pingcap/errors"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-operator/v4/service/controller/blobclient"
+	"github.com/giantswarm/azure-operator/v4/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	azureMachinePool, err := key.ToAzureMachinePool(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	credentialSecret, err := r.getCredentialSecret(ctx, &azureMachinePool)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	blobName := key.BlobName(&azureMachinePool, key.PrefixWorker())
+
+	var payload string
+	{
+		payload, err = r.getCloudConfigFromBootstrapSecret(ctx, azureMachinePool)
+		if errors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "bootstrap CR or cloudconfig secret were not found")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		} else if IsBootstrapCRNotReady(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "bootstrap CR is not ready")
+			r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
+			return nil
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	var containerURL azblob.ContainerURL
+	{
+		containerURL, err = r.getContainerURL(ctx, credentialSecret, key.ClusterID(&azureMachinePool), key.StorageAccountName(&azureMachinePool))
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensuring container object %#q", blobName))
+
+		_, err = blobclient.PutBlockBlob(ctx, blobName, payload, &containerURL)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured container object %#q", blobName))
+	}
+
+	return nil
+}
+
+func (r *Resource) getCloudConfigFromBootstrapSecret(ctx context.Context, azureMachinePool v1alpha3.AzureMachinePool) (string, error) {
+	var sparkCR corev1alpha1.Spark
+	{
+		err := r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: azureMachinePool.Namespace, Name: azureMachinePool.Name}, &sparkCR)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+
+		if !sparkCR.Status.Ready || sparkCR.Status.DataSecretName == "" {
+			return "", microerror.Mask(bootstrapCRNotReady)
+		}
+	}
+
+	var cloudconfigSecret corev1.Secret
+	{
+		err := r.ctrlClient.Get(ctx, client.ObjectKey{Namespace: azureMachinePool.Namespace, Name: sparkCR.Status.DataSecretName}, &cloudconfigSecret)
+		if err != nil {
+			return "", microerror.Mask(err)
+		}
+	}
+
+	return cloudconfigSecret.String(), nil
+}
+
+func (r *Resource) getContainerURL(ctx context.Context, credentialSecret *v1alpha1.CredentialSecret, resourceGroupName, storageAccountName string) (azblob.ContainerURL, error) {
```

Comment:
This function is copy paste code from the `containerurl` handler that was saving the `containerURL` in the controller context.